libraries/AP_HAL/utility/Socket.cpp

```diff
@@ -106,11 +115,30 @@ ssize_t SocketAPM::send(void *buf, size_t size)
 /*
   send some data
  */
-ssize_t SocketAPM::sendto(void *buf, size_t size, const char *address, uint16_t port)
+ssize_t SocketAPM::sendto(const void *buf, size_t size, const char *address, uint16_t port, uint32_t timeout)
```

Comment:
I don't mind having a sendto() that takes a timeout, but the timeout needs to be optional, either with a new API or a convention that timeout=0 means not calling the poll()